### PR TITLE
Add intermediate and root verify flags

### PR DESCRIPTION
**Pull request by malancas**
#### Summary

Part of https://github.com/sigstore/timestamp-authority/issues/121

#### Release Note

Add `--root-certificates` and `--intermediate-certificates` flags to the verify command. The client can now either pass intermediate and root certificates together using the `--certificate-chain` flag or separately using these two new flags.

### CODE REVIEW

File: pkg/tests/cli_test.go [hunk 1]
```diff
@@ -90,14 +90,80 @@ func TestVerify(t *testing.T) {
...
+	pemFiles := writeCertChainToPEMFiles(t, restapiURL)
 
 	// It should verify timestamp successfully.
-	out := runCli(t, "--timestamp_server", restapiURL, "verify", "--timestamp", tsrPath, "--artifact", artifactPath, "--certificate-chain", certChainPemPath, "--nonce", nonce.String(), "--oid", policyOID.String(), "--common-name", commonName)
+	out := runCli(t, "--timestamp_server", restapiURL, "verify", "--timestamp", tsrPath, "--artifact", artifactPath, "--certificate-chain", pemFiles.certChainPath, "--nonce", nonce.String(), "--oid", policyOID.String(), "--common-name", commonName)
 	outputContains(t, out, "Successfully verified timestamp")
 }
 
-func TestVerifyPassLeafCertificate(t *testing.T) {
+func TestVerify_RootAndIntermediateCertificateFlags(t *testing.T) {
```
Comment (malancas) [inline pkg/tests/cli_test.go:100]: I think we can rewrite all or most of these verify tests as table tests in a follow up PR.

File: cmd/timestamp-cli/app/verify.go [hunk 2]
```diff
@@ -144,28 +145,69 @@ func getNonce() (*big.Int, error) {
...
-		return nil, nil, nil
+	rootPEM := viper.GetString("root-certificates")
+	intermediatePEM := viper.GetString("intermediate-certificates")
+
+	// the verify flag must be called with either one of the two flag combinations:
+	// 1. Called with both the --root-certificates flag and the --intermediate-certificates flag
+	// 2. Called with only the --certificate-chain flag
+
+	// this early exit if statement is only entered if neither of those combinations is valid
+	if !((intermediatePEM != "" && rootPEM != "" && certChainPEM == "") || (intermediatePEM == "" && rootPEM == "" && certChainPEM != "")) {
```
[the remark this answers is not in the document]
Comment (malancas) [reply cmd/timestamp-cli/app/verify.go:156]: Ah, that's right

File: cmd/timestamp-cli/app/verify.go [hunk 3]
```diff
@@ -46,11 +46,12 @@ func addVerifyFlags(cmd *cobra.Command) {
 	cmd.Flags().Var(NewFlagValue(fileFlag, ""), "timestamp", "path to timestamp response to verify")
 	cmd.MarkFlagRequired("timestamp") //nolint:errcheck
 	cmd.Flags().Var(NewFlagValue(fileFlag, ""), "certificate-chain", "path to file with PEM-encoded certificate chain. Ordered from intermediate CA certificate that issued the TSA certificate, ending with the root CA certificate.")
-	cmd.MarkFlagRequired("certificate-chain") //nolint:errcheck
 	cmd.Flags().String("nonce", "", "optional nonce passed with the request")
 	cmd.Flags().Var(NewFlagValue(oidFlag, ""), "oid", "optional TSA policy OID passed with the request")
 	cmd.Flags().String("common-name", "", "expected leaf certificate subject common name")
 	cmd.Flags().Var(NewFlagValue(fileFlag, ""), "certificate", "path to file with PEM-encoded leaf certificate")
+	cmd.Flags().Var(NewFlagValue(fileFlag, ""), "intermediate-certificates", "path to file with PEM-encoded intermediate certificates. Must be called with the root-certificate flag.")
+	cmd.Flags().Var(NewFlagValue(fileFlag, ""), "root-certificates", "path to file with a PEM-encoded root certificates. Must be called with the intermediate-certificates flag.")
```
Comment (haydentherapper) [inline cmd/timestamp-cli/app/verify.go:54]:
```suggestion
	cmd.Flags().Var(NewFlagValue(fileFlag, ""), "root-certificates", "path to file with a PEM-encoded root certificates. Optionally can be called with the intermediate-certificates flag.")
```

File: cmd/timestamp-cli/app/verify.go [hunk 4]
```diff
@@ -144,28 +145,77 @@ func getNonce() (*big.Int, error) {
...
+	rootPEM := viper.GetString("root-certificates")
+	intermediatePEM := viper.GetString("intermediate-certificates")
+
+	// the verify flag must be called with either one of the two flag combinations:
+	// 1. Called with the --root-certificates flag and optionally the --intermediate-certificates flag
+	// 2. Called with only the --certificate-chain flag
+
+	// this early exit if statement is only entered if neither of those combinations is valid
+	if !((rootPEM != "" && certChainPEM == "") || (intermediatePEM == "" && rootPEM == "" && certChainPEM != "")) {
+		return nil, nil, fmt.Errorf("the verify command must be called with either only the --certificate-chain flag or with the --root-certificate and --intermediate-certificates flags")
```
Comment (haydentherapper) [inline cmd/timestamp-cli/app/verify.go:157]:
```suggestion
		return nil, nil, fmt.Errorf("the verify command must be called with either only the --certificate-chain flag or with the --root-certificates and --intermediate-certificates flags")
```

**Review by haydentherapper — APPROVED** [review]
Looks great, just comment nits!

**Review by haydentherapper — DISMISSED** [review]

**Review by haydentherapper — APPROVED** [review]

**Pull request merged**